lib/mastalk.rb

```diff
@@ -14,14 +14,15 @@ def initialize(source)
       @source = source.dup
     end
 
-    def to_html
-      ::HTMLEntities.new.decode( kramdown.to_html )
+    def to_html(options = {})
+      options[:auto_id] = options[:auto_id].nil? ? true : options[:auto_id]
+      ::HTMLEntities.new.decode( kramdown(options).to_html() )
     end
 
     private
 
-    def kramdown
-      Kramdown::Document.new(preprocess(source))
+    def kramdown(options)
+      Kramdown::Document.new(preprocess(source), :auto_ids => options[:auto_id])
```

Comment:
Bit repetetive, but the `options[:auto_id] = options[:auto_id].nil? ? true : options[:auto_id]` part should be repeated here, as well as setting the default value in the method signature (`options = {}`).